src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1108,7 +1107,6 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 						shellLaunchConfig.args = undefined;
 					}
 					shellLaunchConfig.executable = await this._resolveVariable(variableResolver, shellOptions.executable);
-					shellSpecified = true;
```

Comment:
why do you believe we don't need to check this anymore?

Reply:
We checked all known shells and added the corresponding command flag on Windows and leave it blank for unknown shells

For other OSs I do think we should check for known shells (instead of always adding `-c`) as well but we're already not checking it for custom terminal profiles anyway... https://github.com/microsoft/vscode/pull/181760#issuecomment-1541566202

Reply:
@alexr00 thoughts here?

Reply:
also @Tyriar ?

Reply:
i'm still concerned about passing in args for a user when they can do that themselves

Reply:
I don't think I'm knowledgable enough to know about this, it certainly looks like a risky change to me but it might be fine.

Reply:
@meganrogge I don't remember the context for this, but it looks like the idea behind this was to only add the arguments if the user didn't include arguments in tasks.json.

Reply:
Yes @Legend-Master